lib/Controller/DirectSessionController.php

```diff
@@ -51,29 +51,23 @@
 use OCP\AppFramework\Controller;
 use OCP\AppFramework\Http\Response;
 use OCP\DirectEditing\IManager;
-use OCP\Share\IManager as ShareManager;
 use OCP\AppFramework\Http\DataResponse;
 use OCP\IRequest;
-use OCP\Share\IShare;
+use OCP\IUserManager;
+use OCP\IUserSession;
 
 class DirectSessionController extends Controller {
+	private ApiService $apiService;
+	private IManager $directManager;
+	private IUserManager $userManager;
+	private IUserSession $userSession;
 
-	/** @var ShareManager */
-	private $shareManager;
-
-	/** @var IShare */
-	private $share;
-
-	/** @var ApiService */
-	private $apiService;
-	/**  @var IManager */
-	private $directManager;
-
-	public function __construct(string $appName, IRequest $request, ShareManager $shareManager, ApiService $apiService, IManager $directManager) {
+	public function __construct(string $appName, IRequest $request, ApiService $apiService, IManager $directManager) {
```

Comment:
The new parameters below are missing, I believe

Reply:
Thanks! I think i addressed this now. Also merged the lint fix into the other commit.

Reply:
static check seems happy now - and so am i. :wink: